src/Core/Vipr/FileWriter.cs

```diff
@@ -26,8 +31,37 @@ public static void Write(IEnumerable<TextFile> textFilesToWrite, string outputDi
                 if (!Directory.Exists(Path.GetDirectoryName(filePath)))
                     Directory.CreateDirectory(Path.GetDirectoryName(filePath));
 
-                File.WriteAllText(filePath, file.Contents);
+                fileTasks.Add(WriteToDisk(filePath, file.Contents));
             }
+            await Task.WhenAll(fileTasks);
+        }
+
+        /**
+         * Write the file to disk. If the file is locked for editing,
+         * sleep until available
+         */
+        public static async Task WriteToDisk(string filePath, string output)
+        {
+            for (int tries = 0; tries < 10; tries++)
+            {
+                StreamWriter sw = null;
+                try
+                {
+                    using (sw = new StreamWriter(filePath, false, Encoding.UTF8))
+                    {
+                        await sw.WriteAsync(output);
+                        break;
+                    }
+                }
+                // If the file is currently locked for editing, sleep
+                // This shouldn't be hit if the generator is running correctly,
+                // however, files are currently being overwritten several times
+                catch (IOException)
+                {
+                    Thread.Sleep(5);

```

Comment:
Would be better to use a Monitor around this code than sleeping the thread. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/lock-statement

If you are hitting this code often, than a {lock} might have performance implications. We can discuss other patterns if that is the case. 
https://docs.microsoft.com/en-us/dotnet/standard/threading/managed-threading-best-practices